crates/ty_python_semantic/src/semantic_index.rs

```diff
@@ -230,6 +280,9 @@ pub(crate) struct SemanticIndex<'db> {
     /// The set of modules that are imported anywhere within this file.
     imported_modules: Arc<FxHashSet<ModuleName>>,
 
+    /// The set of modules that are imported anywhere within this file.
+    maybe_imported_modules: Arc<FxHashSet<MaybeModuleImport>>,
```

Comment:
I don't think this needs to be an `Arc`, as we have no `#[salsa::tracked]` query that returns the maybe imported modules directly

```suggestion
    maybe_imported_modules: FxHashSet<MaybeModuleImport>,
```